src/main/java/io/github/syst3ms/skriptparser/util/StringUtils.java

```diff
@@ -324,22 +324,79 @@ else if (plural)
         return expr.getValues(ctx);
     }
 
-    public static String camelCase(String str, boolean firstNoCase) {
+    /**
+     * Capitalizes the first word or all words in a string.
+     * A word is separated by a space character.
+     * @param str the string to change
+     * @return the capitalized string
+     */
+    public static String toTitleCase(String str, boolean allWords) {
+        if (!allWords) {
+            return Character.toUpperCase(str.charAt(0)) + str.substring(1);
+        } else {
+            String[] words = str.split(" ");
+
+            for (int i = 0; i < words.length; i++) {
+                words[i] = Character.toUpperCase(words[i].charAt(0)) + words[i].substring(1);
+            }
+            return String.join(" ", words);
+        }
+    }
+
+    public static String toCamelCase(String str, boolean firstNoCase) {
         String[] parts = str.split("\\s+");
         StringBuilder ret = new StringBuilder();
         for (String part : parts) {
             if (firstNoCase) {
                 firstNoCase = false;
-                ret.append(part.toLowerCase());
+                ret.append(Character.toLowerCase(part.charAt(0)))
+                        .append(part.substring(1));
             } else {
                 ret.append(Character.toUpperCase(part.charAt(0)))
-                        .append(part.substring(1).toLowerCase());
+                        .append(part.substring(1));
             }
         }
         return ret.toString();
     }
```

Comment:
If a skript user try to use a unicode caracter in a string who be converted as title case or as camel case, does Skript will return to it a Java Error ? due to non-handle of theses caracters

Reply:
I did not think of this, since I mainly copied the code from the Skript repository and enhanced it where necessary (some of the methods were there already).
I will test it but my personal guess is that it works fine.